### PR TITLE
Issue 6441 - Add basic dsidm posixgroup tests

### PR DESCRIPTION
**Pull request by mirielka**
New basic dsidm posixgroup test - create, list, get, get_dn, modify, rename, rename with keep-old-rdn option.

Relates: https://github.com/389ds/389-ds-base/issues/6441

Author: Lenka Doudova

Reviewed by: ???

### CODE REVIEW

File: dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+
+    log.info('Test empty dsidm posixgroup list without json')
+    args.json = False
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, check_value_not=posixgroup_name)
+
+
+@pytest.mark.xfail(reason="Will fail because of DS6439")
+@pytest.mark.skipif(ds_is_older("1.4.2"), reason="Not implemented")
+def test_dsidm_group_get_dn(topology_st, create_test_group):
```
Comment (droideck) [inline dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py]: Probably, it makes sense to rename the test function to `test_dsidm_posixgroup_get_dn` for consistency

File: dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+    args.gidNumber = '3000'
+
+    log.info('Test dsidm posixgroup create')
+    create(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, check_value=output)
+
+    log.info('Check that posixgroup is present')
+    posixgroups = PosixGroups(standalone, DEFAULT_SUFFIX)
+    new_posixgroup = posixgroups.get(posixgroup_name)
+    assert new_posixgroup.exists()
```
Comment (droideck) [inline dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py:89]: I think we should also check that the group has previously specified `args.gidNumber = '3000'`

File: dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+    posixgroups = PosixGroups(standalone, DEFAULT_SUFFIX)
+    test_posixgroup = posixgroups.get(posixgroup_name)
+
+    posixgroup_content = ['dn: {}'.format(test_posixgroup.dn),
+                    'cn: {}'.format(test_posixgroup.rdn),
+                    'gidNumber: 3000',
+                    'objectClass: top',
+                    'objectClass: groupOfNames',
+                    'objectClass: posixGroup',
+                    'objectClass: nsMemberOf']
```
Comment (droideck) [inline dirsrvtests/tests/suites/clu/dsidm_posixgroup_test.py]: Alignment

**Review by droideck — CHANGES_REQUESTED** [review]
Couple of minor issue, but overall - looks great!

**Comment by mirielka** [comment]
Thanks for review, fixes pushed

**Comment by mirielka** [comment]
Added fix for failing get_dn option.

**Review by droideck — APPROVED** [review]
Looks good! Ack

**Pull request merged**